### PR TITLE
Add NumPyDoc docstring to BaseCursor.execute

### PR DESCRIPTION
**Pull request by MartinThoma**
* Parameters explained
* Return value explained

NumPyDoc is a standard docstring format (see [guide](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt)) which leads is very well readable in the source code, in the `help()`. It can also be used to generate beautiful documentation.

See the following examples  [`Lasagne/lasagne/layers/input.py`](https://github.com/Lasagne/Lasagne/blob/master/lasagne/layers/input.py) and [website docs](http://lasagne.readthedocs.org/en/latest/modules/layers.html)

### CODE REVIEW

**Comment by MartinThoma** [comment]
I've just created this docstring (1) to give an example how it can be used (2) I wanted to know the return value of `cursor.execute` and it was not obvious.

**Comment by MartinThoma** [comment]
Related to #199

**Comment by methane** [comment]
I prefer sphinx's docstring format since sphinx autodoc supports it by default.

File: pymysql/cursors.py [hunk 1]
```diff
@@ -137,7 +137,25 @@ def mogrify(self, query, args=None):
...
     def execute(self, query, args=None):
-        '''Execute a query'''
+        """
+        Execute a query
+
+        Parameters
+        ----------
+        query : str
+            Query to execute on the server.
+        args : sequence or mapping, optional
```
Comment (methane) [inline pymysql/cursors.py]: `tuple, list or dict` since `_escape_args()` only supports them.

File: pymysql/cursors.py [hunk 2]
```diff
@@ -137,7 +137,25 @@ def mogrify(self, query, args=None):
...
+            Will be used with query
+
+        Returns
+        -------
+        int
+            Number of affected rows
+
+        Notes
+        -----
+        If args is a sequence, %s can be used as a placeholder in the query.
```
Comment (methane) [inline pymysql/cursors.py]: If args is a dict, %(name)s can be used.

**Comment by MartinThoma** [comment]
@methane It is relatively easy to add numpydoc to sphinx (see https://codeandchaos.wordpress.com/2012/08/09/sphinx-and-numpydoc/) and numpydoc is MUCH easier to read.

Is there any official documentation generated by Sphinx by now? If not, I think it only matters how easier it is to read it in the command line with `help()`.

**Comment by methane** [comment]
Why I prefer sphinx style is:

* Out of the box is better than plugin required.
* I know jedi and PyCharm supports sphinx style docstring.  I don't know how about NumPy syntax.

> Is there any official documentation generated by Sphinx by now?

mysqlclient-python has readthedocs site.  Currently, autodoc doesn't work but I want to support it in future.

**Comment by methane** [comment]
There is also google style.
I like one good style than 10 better styles.

**Comment by methane** [comment]
I researched quickly.

https://www.jetbrains.com/pycharm/help/python-integrated-tools.html
PyCharm 5 adds support for NumPy and Google style format. good.

http://jedi.jedidjah.ch/en/latest/docs/features.html#type-hinting
Jedi supports NumPy format, but it requires numpydoc plugin.
It means common installation of [jedi-vim](davidhalter/jedi-vim) doesn't support numpy syntax. -1.

**Comment by MartinThoma** [comment]
@methane I like documentation at all better than no documentation. I've added a bit of documentation where nothing was before.

I think the discussion about the documentation style should be in #199 as it is a general question.

**Comment by methane** [comment]
merged

**Pull request closed without merging**